### PR TITLE
libct/devices: change devices.Type to be a string

### PR DESCRIPTION
**Pull request by thaJeztah**
> I noticed this while reviewing https://github.com/opencontainers/runc/pull/3110. Opening as draft to see if there's any failures in CI, and because this will conflict with that PR

Possibly there was a specific reason to use a rune for this, but I noticed
that there's various parts in the code that has to convert values from a
string to this type. Using a string as type for this can simplify some of
that code.

### CODE REVIEW

**Comment by thaJeztah** [comment]
One failure; something I need to look into

```
=== RUN   TestDeviceFilter_Weird
299
    devicefilter_test.go:26: TestDeviceFilter_Weird: unable to remove 'allow' exception: requested rule [{b 8 2} rwm] not supported by devices cgroupv1 (cannot punch hole in existing wildcard rule [{b -1 -1} rwm]) (devices: [0xc000012e40 0xc000012e80 0xc000012ec0])
300
--- FAIL: TestDeviceFilter_Weird (0.00s)
```

**Comment by kolyshkin** [comment]
I would use `byte` as it's a single character. String is OKish

**Comment by thaJeztah** [comment]
> I would use `byte` as it's a single character

Doing so would bring back the same/similar situation as `rune`. The OCI type uses `string` for the `Type` field (and I guess JSON also doesn't have a concept of `byte` or `rune` values, other than using a `Number` for them).

**Review by kolyshkin — CHANGES_REQUESTED** [review]
Can you split the removal of `stringToCgroupDeviceRune` to a separate commit please?

**Comment by thaJeztah** [comment]
> Can you split the removal of `stringToCgroupDeviceRune` to a separate commit please?

Just the removal, or also calling the function?

I'll have a look tomorrow if I think of doing so 😁

**Comment by kolyshkin** [comment]
> Just the removal, or also calling the function?

I was thinking that a separate first commit in which `stringToCgroupDeviceRune` (together with its use) is eliminated would make sense, but now I see doing this would require a temporary use of `stringToDeviceType` with an added error check, so maybe it's not worth it.

**Review by kolyshkin — APPROVED** [review]
LGTM

@cyphar PTAL

**Comment by thaJeztah** [comment]
I think this one may conflict with https://github.com/opencontainers/runc/pull/3110, do you want me to keep it in draft until after that's merged? Otherwise I can move this out of draft now

**Review by cyphar — APPROVED** [review]
LGTM. I'm not sure but I think the usage of runes came from the idea that we should use the equivalent of `char` because "that's what you'd do in C", even though in Go `string`s are far more convenient than `rune` or `byte`.

**Comment by thaJeztah** [comment]
> I think the usage of runes came from the idea that we should use the equivalent of char because "that's what you'd do in C", even though in Go strings are far more convenient than rune or byte.

Yes, makes sense! (I was wondering if there was (e.g.) some performance reason behind it that I overlooked)

I moved this one out of draft, but happy to rebase if https://github.com/opencontainers/runc/pull/3110 is gonna be merged first.

**Comment by cyphar** [comment]
@kolyshkin Feel free to merge this if you don't think the rebase of #3110 wouldn't be too bad.

**Pull request merged**
